_posts/2020-03-31-bazel-3.0.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: posts
+title: "Bazel 3.0"
+authors:
+  - katre
+---
+
+[Bazel 3.0](https://github.com/bazelbuild/bazel/releases/tag/3.0.0) is a new major release and brings the following incompatible changes:
+
+## Incompatible Changes
+
+The following incompatible flags are now enabled by default:
+
+*   [`--incompatible_remove_ram_utilization_factor`](https://github.com/bazelbuild/bazel/issues/10730)
+*   [`--incompatible_remove_local_resources`](https://github.com/bazelbuild/bazel/issues/10536)
+*   [`--incompatible_load_proto_toolchain_for_javalite_from_com_google_protobuf`](https://github.com/bazelbuild/bazel/issues/10335)
+*   [`--incompatible_always_check_depset_elements`](https://github.com/bazelbuild/bazel/issues/10313)
+*   [`--incompatible_remove_enabled_toolchain_types`](https://github.com/bazelbuild/bazel/issues/10262)
+*   [`--incompatible_validate_top_level_header_inclusions`](https://github.com/bazelbuild/bazel/issues/10047)
+
+Before upgrading to Bazel 3.0, you can test your build with Bazel 2.2 and the flags above. We expect that the migration to Bazel 3.0 will be simple for most users. Please comment on the corresponding incompatible flag issue if any of the flags causes migration problems, or open a [new issue](http://github.com/bazelbuild/bazel/issues/new) if you have other problems.
+
+### New Incompatible Changes
+
+*   Adds [`--incompatible_objc_compile_info_migration`](https://github.com/bazelbuild/bazel/issues/10854) to indicate whether native Objective C/C++ rules should assume compile info has been migrated to CcInfo.
+
+## Java
+
+*   Using JDK 9 or 10 as a `--host_javabase` is no longer officially supported. As always, you can use the `@bazel_tools//tools/jdk:toolchain_vanilla` Java toolchain to use older or newer JDKs than what Bazel currently supports.
+
+## C++
+
+*   Introduce flag [`--incompatible_require_linker_input_cc_api`](https://github.com/bazelbuild/bazel/issues/10860) which changes the C++ Starlark API for linking contexts, libraries to link and linker inputs. Not flipped in this release.
+
+## Other important changes
+
+*   `--fatal_event_bus_exceptions` is deprecated and should not be used. Any crashes should be reported so that they can be fixed.
```

Comment:
```suggestion
*   `--fatal_event_bus_exceptions` is deprecated and should not be used. Any crashes should be [reported](https://github.com/bazelbuild/bazel/issues/new) so that they can be fixed.
```